scipy/sparse/csgraph/_flow.pyx

```diff
@@ -622,5 +656,5 @@ cdef ITYPE_t[:] _dinic(
             flow = _augment_paths(edge_ptr, source, sink,
```

Comment:
As the resulting flow is not used for anything, you could let `_augment_paths` return a boolean, thus avoiding the `flow = 1; while flow` here.